File: backend/src/main/kotlin/com/trackwatch/backend/model/Artist.kt
package com.trackwatch.backend.model

import jakarta.persistence.Column
import jakarta.persistence.Embeddable

@Embeddable
data class Artist(
    @Column(name = "artist_id")
    val id: String,

    @Column(name = "artist_name")
    val name: String,

    val imageUrl: String = ""
) {
    fun isEqualTo(other: Any): Boolean {
        if (other !is Artist) {  return false  }
        return this.id == other.id  && this.name == other.name
    }
}

File: backend_py/trackwatch/app/models/artist.py
class Artist:
  def __init__(self, id, name, image_url=""):
      self.id = id
      self.name = name
      self.image_url = image_url

  def is_equal_to(self, other):
    if not isinstance(other, Artist): return False
    return self.id == other.id and self.name == other.name